src/OpenTelemetry.Abstractions/OpenTelemetry.Abstractions.csproj

```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net45;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net46;netstandard2.0</TargetFrameworks>
```

Comment:
if the only purpose to change target is Task.CompletedTask I'd prefer to copy-paste implementation from 
https://referencesource.microsoft.com/#mscorlib/system/threading/Tasks/Task.cs,1650 and keep net45.

Reply:
Right, I've done that before, it's not a big deal. 
I wanted to bring this up because 4.5 might limit us in the future (i.e: has no `AsyncLocal` so I was wondering if we really need to support 4.5 as opposed to 4.6).
In any case, happy to make a `Task.CompletedTask`.

Reply:
All propagation happens in the sdk which targets net46. It's just if someone wants to implement sdk for net45 they can do it. or if any library targets net45 - they don't need to introduce new targets and ifdef

Reply:
Unfortunately the implementation of `Task.CompletedTask` uses an internal constructor. The approach I've used before was to use `Task.WhenAll()` instead, which creates a new completed successfully task, synchronously. We can either just return that (and have a single `#if NET45` or create a class class an internal like `TaskShim` and `#ifdef` the class/import and call to `TaskShim.CompletedTask` in the code so we can cache the instance. Or something else you prefer?

Reply:
Can you just use `Task.FromResult` ? If you need void, you could do [this](https://stackoverflow.com/questions/13127177/if-my-interface-must-return-task-what-is-the-best-way-to-have-a-no-operation-imp)